lua/telescope/actions/init.lua

```diff
@@ -272,6 +272,16 @@ function actions.close_pum(_)
   end
 end
 
+--- Copy current entry command to unnamed plus register
+---@param prompt_bufnr number: The prompt bufnr
+function actions.copy_command_to_reg(prompt_bufnr)
```

Comment:
Maybe we should make the register customizable by adding a `reg` parameter that defaults to `+` because (i) users might want to have it in another reg to begin with and (ii) [os differences](https://stackoverflow.com/questions/30691466/what-is-difference-between-vims-clipboard-unnamed-and-unnamedplus-settings)

Reply:
One way of doing this is to put a configurable variant in `actions.generate` with the `which_key` example.

Reply:
I will try to understand what `actions.generate` do later, I did one attempt using fn.input, it's working.

Reply:
I still have a question, where would the user override it?

Reply:
I think as it is, the user has to select a register via `vim.fn.input` (by either keeping it empty or not). What I had in mind is --

```lua
function actions.copy_command_to_reg(prompt_bufnr, opts)
  opts = opts or {} -- ensure opts is a table if nil
  opts.reg = utils.get_default(opts.reg, "+")
  local cmd = action_state.get_selected_entry()["name"]
  -- concat ':' with command name entry
  a.fn.setreg(opts.reg, ":" .. cmd, "c")
end
```
and then later in `actions.generate`

```lua
action_generate.copy_command_to_reg(opts)
  return function(prompt_bufnr)
     action.copy_command_to_reg(prompt_bufnr, opts)
  end
end
```

And then in `telescope.setup{}`, users would map `action_generate.copy` which returns the customized action.

Reply:
So in our idea if I wanted to change the reg from the default one, I have to do when mapping like:
`["C-y"] = actions.copy_command("a") ?`